lib/fairy/fairy.c

```diff
@@ -314,30 +335,30 @@ void Fairy_InitFile(FairyFileInfo* fileInfo, FILE* file) {
                     }
                     break;
 
+                case SHT_RELA:
+                    off += 1;
                 case SHT_REL:
+                    off += 5;
                     /* This assumes only one reloc section of each name */
                     // TODO: is this a problem?
```

Comment:
I'm not sure if it will ever come up in practice, but in principle multiple relocation sections can exist for a given progbits section. You can find what section a relocation section relocates using the `sh_info` field. This field for `SHT_REL` and `SHT_RELA` sections contains the section number of the section that it relocates.
I think it's worth keeping the comment and implementing it if it's ever discovered in the wild (which it hopefully won't be but who can say)

Reply:
Yeah, that's fair enough, we've seen all manner of things from GCC so far.